numpy/core/src/multiarray/array_method.c

```diff
@@ -819,39 +817,37 @@ unmasked_ufunc_loop_as_masked(
 
         /* Process unmasked values */
         mask = npy_memchr(mask, 0, mask_stride, N, &subloopsize, 0);
-        strided_loop(dataptrs, &subloopsize, strides, innerloopdata);
+        int res = strided_loop(context,
+                dataptrs, &subloopsize, strides, strided_loop_auxdata);
+        if (res != 0) {
+            return res;
+        }
         for (int i = 0; i < nargs; i++) {
             dataptrs[i] += subloopsize * strides[i];
         }
         N -= subloopsize;
     } while (N > 0);
+
+    return 0;
 }
 
 
 /*
- * TODO: This function will be the masked equivalent to `get_loop`.
- * This function wraps a legacy inner loop so it becomes masked.
- *
- * Returns 0 on success, -1 on error.
+ * Identical to the `get_loop` functions and wraps it.  This adds support
+ * to a boolean mask being passed in as a last, additional, operand.
```

Comment:
Not clear: where is the boolean mask passed in. Do we even support boolean masks? Maybe leave this wrapper for a future PR when we add that support